src/main/java/org/spongepowered/api/util/Color.java

```diff
@@ -38,135 +38,179 @@
-     * color. Note that colors do not utilize an alpha modifier
+     * Creates a new {@link Color} based on the red, green, and blue values.
      *
      * @param red The red value
      * @param green The green value
      * @param blue The blue value
      * @return The color object
      */
-    public static Color ofRgb(int red, int green, int blue) {
-        return new Color(red, green, blue);
+    public static Color of(int red, int green, int blue) {
+        //TODO: Remove checkArguments and use (red & 0xFF)?
+        checkArgument(0 <= red && red <= 0xFF, "red value %s (%s) must be in range 0 to 0xFF (255)", Integer.toHexString(red), red);
+        checkArgument(0 <= green && green <= 0xFF, "green value %s (%s) must be in range 0 to 0xFF (255)", Integer.toHexString(green), green);
+        checkArgument(0 <= blue && blue <= 0xFF, "blue value %s (%) must be in range 0 to 0xFF (255)", Integer.toHexString(blue), blue);
+        return new Color((red << 16) | (green << 8) | blue, red, green, blue);
     }
 
     /**
-     * Converts the provided {@link java.awt.Color} object into a valid
-     * {@link Color} object to be used throughout the API.
+     * Creates a new {@link Color} based on the red, green, and blue values.
      *
-     * @param color The java color object
-     * @return The converted color object
+     * @param red The red value
+     * @param green The green value
+     * @param blue The blue value
+     * @return The color object
+     * @deprecated in favor of {@link #of(int, int, int)}
      */
-    public static Color of(java.awt.Color color) {
-        return new Color(color.getRed(), color.getGreen(), color.getBlue());
+    @Deprecated
+    public static Color ofRgb(int red, int green, int blue) {
+        return of(red, green, blue);
     }
 
     /**
-     * Converts the provided {@link Vector3i} into a {@link Color} object.
+     * Creates a new {@link Color} based on the x (red), y (green), and z (blue)
+     * values of the vector.
      *
      * @param vector3i The vector of three integers representing color
      * @return The color object
      */
     public static Color of(Vector3i vector3i) {
-        return new Color(vector3i.getX(), vector3i.getY(), vector3i.getZ());
+        return of(vector3i.getX(), vector3i.getY(), vector3i.getZ());
     }
 
     /**
-     * converts the provided {@link Vector3f} into a {@link Color} object.
+     * Creates a new {@link Color} based on the x (red), y (green), and z (blue)
+     * values of the vector. Values are rounded to the nearest whole number.
      *
      * @param vector3f The vector of three floats representing color
      * @return The color object
+     * @deprecated Use {@code #of(vector3f.round().toInt())}
      */
+    @Deprecated
     public static Color of(Vector3f vector3f) {
-        return new Color(Math.round(vector3f.getX()), Math.round(vector3f.getY()), Math.round(vector3f.getZ()));
+        return of(vector3f.round().toInt());
```

Comment:
I agree, it's a ridiculous way of handling it which is why I deprecated the method. I like the idea of interpreting this as 0-1 values as I know of at least one place in Sponge where this is used and it's how Mojang stores colors as well. This would change the method to have a different result, but I can probably check the vector for values between 0 and 1 and handle it from there. Thoughts?